llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -301,15 +300,17 @@ static unsigned getShufflevectorNumGroups(ArrayRef<Value *> VL) {
   auto *SV = cast<ShuffleVectorInst>(VL.front());
   unsigned SVNumElements =
       cast<FixedVectorType>(SV->getOperand(0)->getType())->getNumElements();
-  unsigned GroupSize = SVNumElements / SV->getShuffleMask().size();
+  unsigned ShuffleMaskSize = SV->getShuffleMask().size();
+  unsigned GroupSize = SVNumElements / ShuffleMaskSize;
   if (GroupSize == 0 || (VL.size() % GroupSize) != 0)
     return 0;
   unsigned NumGroup = 0;
   for (size_t I = 0, E = VL.size(); I != E; I += GroupSize) {
     auto *SV = cast<ShuffleVectorInst>(VL[I]);
     Value *Src = SV->getOperand(0);
     ArrayRef<Value *> Group = VL.slice(I, GroupSize);
-    SmallVector<int> ExtractionIndex(SVNumElements);
+    SmallVector<int> ExpectedIndex(
+        createStrideMask(0, ShuffleMaskSize, GroupSize));
```

Comment:
You can still use bitvector, just mark not elements in the groups, but the groups themselves.

Reply:
Fixed.